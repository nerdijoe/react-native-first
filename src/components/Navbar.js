import React from 'react'
import {
  View,
  Text,
  TouchableOpacity
} from 'react-native'
import { Actions } from 'react-native-router-flux'

const styles = {
  navbar: {
    margin: 0,
    padding: 10,
    backgroundColor: '#EDB1F1',
    flexDirection: 'row',
    justifyContent: 'space-around'
  }
}

class Navbar extends React.Component {

  onPressHandler = (key) => {
    console.log("onPress " + key)
    if(key === 'cards')
      Actions.cards()
    else
      Actions.arenas()
  }

  render() {
    return (
      <View style={styles.navbar}>
        <TouchableOpacity active={true} onPress={() => {this.onPressHandler('cards')}}>
          <Text>Cards</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => {this.onPressHandler('arenas')}}>
          <Text>Arenas</Text>
        </TouchableOpacity>
      </View>
    )
  }
}

export default Navbar
