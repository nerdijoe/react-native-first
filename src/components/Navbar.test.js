import React from 'react'
import renderer from 'react-test-renderer'

import Navbar from './Navbar'


describe ('Navbar', () => {
  it('render without crashing', () => {
    const tree = renderer.create(<Navbar />)

    expect(tree).toMatchSnapshot()
  })
})
