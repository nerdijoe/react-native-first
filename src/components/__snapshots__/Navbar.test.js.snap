// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar render without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#EDB1F1",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": 0,
      "padding": 10,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      カード
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      カード2
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      アリーナ
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      カメラ
    </Text>
  </View>
</View>
`;
